Libraries/DirectMLX.h

```diff
@@ -3470,9 +3475,48 @@ namespace dml
         return out;
     }
 
-    // 
-    // TODO: NonZeroCoordinates
-    // 
+    struct NonZeroCoordinatesOutputs
+    {
+        Expression count;
+        Expression coordinates;
+    };
+    inline NonZeroCoordinatesOutputs NonZeroCoordinates(Expression input)
+    {
+        detail::GraphBuilder* builder = input.Impl()->GetGraphBuilder();
+        TensorDesc inputTensor = input.Impl()->GetOutputDesc();
+        const auto& inputTensorSizes = inputTensor.sizes;
+        uint32_t dimensionCount = static_cast<uint32_t>(inputTensorSizes.size());
+
+        TensorDimensions outputSizes = {};
+        uint32_t effectiveRank = -1;
+        uint32_t totalElements = 1;
+        for (uint32_t i = 0; i < dimensionCount; ++i)
+        {
+            if (inputTensorSizes[i] > 1 && effectiveRank == -1)
+            {
+                effectiveRank = dimensionCount - i;
+            }
```

Comment:
If all values in `inputTensorSizes` are 1, wouldn't `effectiveRank` have a value of `UINT32_MAX`?